scripts/biomedical/VirtualMetabolicHuman/format_metabolite.py

```diff
@@ -0,0 +1,475 @@
+        vmh_id = 'NA'    
+
+    writer.writerow({'accession': accession, 'monisotopic_molecular_weight': monisotopic_molecular_weight, 'iupac_name': iupac_name, 'name': name, 'chemical_formula': chemical_formula, 'InChIKey': inchikey, 'cas_registry_number': cas_registry_number, 'smiles': smiles,'drugbank': drugbank,'chebi_id': chebi_id,'pubchem': pubchem,'phenol_explorer_compound_id':phenol_explorer_compound_id, 'food': food,'knapsack': knapsack, 'chemspider': chemspider,'kegg': kegg, 'meta_cyc': meta_cyc, 'bigg':bigg, 'metlin_id': metlin_id, 'pdb_id':pdb_id,'logpexp':logpexp, 'kingdom': kingdom, 'direct_parent': direct_parent, 'super_class': super_class, 'class': classorg, 'sub_class': sub_class, 'molecular_framework': molecular_framework, 'vmh_id': vmh_id})
+    # It's safe to call clear() here because no descendants will be
+    # accessed
+    elem.clear()
+    # Also eliminate now-empty references from the root node to elem
+    for ancestor in elem.xpath('ancestor-or-self::*'):
+        while ancestor.getprevious() is not None:
+            del ancestor.getparent()[0]
+  del context
+  return;
+
+#Used to get hmdb id for common name matches between vmh and hmdb
+def name_map(dfm, dfh):
+    dfm['fullName'] = dfm['fullName'].map(lambda x: re.sub(r'\W+', '', x))
+    dfm['fullName'] = dfm['fullName'].str.lower()
+    dfh['name'] = dfh['name'].map(lambda x: re.sub(r'\W+', '', x))
+    dfh['name'] = dfh['name'].str.lower()
+    dfh['name'] = dfh['name'].str[2:]
+    dfh['name'] = dfh['name'].str[:-1]
+    dfh.rename(columns = {'name':'fullName'}, inplace = True)
+    df = pd.merge(dfm, dfh, on = 'fullName', how = 'left')
+    df['hmdb'].fillna(df['accession'])
+    df = df.iloc[:,1:len(dfm.columns)]
+    return df
+
+#Used to get hmdb id for KEGG id matches between vmh and hmdb
+def kegg_map(dfm, dfh):
+    dfh.rename(columns = {'kegg':'keggId'}, inplace = True)
+    df = pd.merge(dfm, dfh, on = 'keggId', how = 'left')
+    df['hmdb'].fillna(df['accession'])
+    df = df.iloc[:,1:len(dfm.columns)]
+    return df
+
+#Used to get hmdb id for CHEBI id matches between vmh and hmdb
+def chebi_map(dfm, dfh):
+    dfh.rename(columns = {'chebi_id':'cheBlId'}, inplace = True)
+    df = pd.merge(dfm, dfh, on = 'cheBlId', how = 'left')
+    df['hmdb'].fillna(df['accession'])
+    df = df.iloc[:,1:len(dfm.columns)]
+    return df
+
+def isNaN(num):
+    return num != num
+
+def clean_result(result):
+    ind_count = 0
+    dcid_inch = []
+    for index in range(len(result)):
+        for key in result[index]:
+            dcid_inch.insert(ind_count, result[index][key])
+            ind_count += 1
+    return dcid_inch
+
+def add_query_result(df, col, dcid):
+    count_query = 0
+    for i in df.index:
+        for j in range(1, len(dcid)):
+            if df.loc[i, col] == dcid[j]:
+                count_query += 1
+                df.loc[i, 'Id'] = dcid[j - 1]
+                j += 2
+    return df
+
+def main():
+    file_input = sys.argv[1]
```

Comment:
For this file and the rest, would be good to document the required arguments in the file header.

Reply:
Done !